fedimint-client/src/transaction/sm.rs

```diff
@@ -146,37 +145,35 @@ impl State for TxSubmissionStatesSM {
 }
 
 impl TxSubmissionStates {
-    async fn trigger_created_rejected(tx: Transaction, context: DynGlobalClientContext) -> String {
+    async fn trigger_created_rejected(
+        transaction: Transaction,
+        context: DynGlobalClientContext,
+    ) -> String {
+        let mut backoff = custom_backoff(Duration::from_secs(2), Duration::from_secs(600), None);
+
         loop {
-            match context.api().submit_transaction(tx.clone()).await {
-                Ok(serde_outcome) => match serde_outcome.try_into_inner(context.decoders()) {
-                    Ok(outcome) => {
-                        if let TransactionSubmissionOutcome(Err(transaction_error)) = outcome {
-                            return transaction_error.to_string();
-                        }
-                    }
-                    Err(decode_error) => {
-                        warn!(target: LOG_CLIENT_NET_API, error = %decode_error, "Failed to decode SerdeModuleEncoding");
+            match context
+                .api()
+                .submit_transaction(transaction.clone())
+                .await
+                .try_into_inner(context.decoders())
+            {
+                Ok(outcome) => {
+                    if let TransactionSubmissionOutcome(Err(transaction_error)) = outcome {
+                        return transaction_error.to_string();
                     }
-                },
-                Err(error) => {
-                    error.report_if_important();
                 }
-            }
+                Err(decode_error) => {
+                    warn!(target: LOG_CLIENT_NET_API, error = %decode_error, "Failed to decode SerdeModuleEncoding");
+                }
+            };
 
-            sleep(RETRY_INTERVAL).await;
+            sleep(backoff.next().expect("Number of retries is has no limit")).await;
         }
     }
 
     async fn trigger_created_accepted(txid: TransactionId, context: DynGlobalClientContext) {
-        loop {
-            match context.api().await_transaction(txid).await {
-                Ok(..) => return,
-                Err(error) => error.report_if_important(),
-            }
-
-            sleep(RETRY_INTERVAL).await;
-        }
+        context.api().await_transaction(txid).await;
```

Comment:
Huh. This is kind of good. We'll never need a higher level retry, and always just use lower level ones.

One thing that I'm worried about is that now it's a bit unclear which methods need a higher level loop and which don't. But I guess `await_...` might be a good enough clue?

Reply:
You see if the return a federation result or not.

Reply:
Huh... Yeah... I guess it's hard to make a retry loop around something that doesn't fail.